File: backend/src/yourai/core/enums.py
"""Shared StrEnum definitions mirroring PostgreSQL enum types.

These enums are the single Python source of truth for all enum values.
They MUST match the PostgreSQL enums defined in DATABASE_SCHEMA.sql.
"""

from enum import StrEnum


class SubscriptionTier(StrEnum):
    STARTER = "starter"
    PROFESSIONAL = "professional"
    ENTERPRISE = "enterprise"


class UserStatus(StrEnum):
    PENDING = "pending"
    ACTIVE = "active"
    DISABLED = "disabled"
    DELETED = "deleted"


class ConversationState(StrEnum):
    PENDING = "pending"
    WAITING_FOR_REPLY = "waiting_for_reply"
    GENERATING_REPLY = "generating_reply"
    OUTPUTTING_REPLY = "outputting_reply"
    READY = "ready"


class MessageRole(StrEnum):
    USER = "user"
    ASSISTANT = "assistant"


class MessageState(StrEnum):
    PENDING = "pending"
    SUCCESS = "success"
    ERROR = "error"
    CANCELLED = "cancelled"


class ConfidenceLevel(StrEnum):
    HIGH = "high"
    MEDIUM = "medium"
    LOW = "low"


class VerificationStatus(StrEnum):
    VERIFIED = "verified"
    UNVERIFIED = "unverified"
    REMOVED = "removed"
    PRE_1963_DIGITISED = "pre_1963_digitised"


class AgentInvocationMode(StrEnum):
    CONVERSATION = "conversation"
    POLICY_REVIEW = "policy_review"


class ModelTier(StrEnum):
    HAIKU = "haiku"
    SONNET = "sonnet"
    OPUS = "opus"


class KnowledgeBaseCategory(StrEnum):
    LEGISLATION = "legislation"
    CASE_LAW = "case_law"
    EXPLANATORY_NOTES = "explanatory_notes"
    AMENDMENTS = "amendments"
    COMPANY_POLICY = "company_policy"
    SECTOR_KNOWLEDGE = "sector_knowledge"
    PARLIAMENTARY = "parliamentary"


class KnowledgeBaseSourceType(StrEnum):
    LEX_API = "lex_api"
    UPLOADED = "uploaded"
    CATALOG = "catalog"
    PARLIAMENT_MCP = "parliament_mcp"


class DocumentProcessingState(StrEnum):
    UPLOADED = "uploaded"
    VALIDATING = "validating"
    EXTRACTING_TEXT = "extracting_text"
    CHUNKING = "chunking"
    CONTEXTUALISING = "contextualising"
    EMBEDDING = "embedding"
    INDEXING = "indexing"
    READY = "ready"
    FAILED = "failed"


class GuardrailStatus(StrEnum):
    CREATING = "creating"
    UPDATING = "updating"
    VERSIONING = "versioning"
    READY = "ready"
    FAILED = "failed"
    DELETING = "deleting"


class ReviewCycle(StrEnum):
    ANNUAL = "annual"
    MONTHLY = "monthly"
    QUARTERLY = "quarterly"


class PolicyReviewState(StrEnum):
    PENDING = "pending"
    PROCESSING = "processing"
    VERIFYING = "verifying"
    COMPLETE = "complete"
    ERROR = "error"
    CANCELLED = "cancelled"


class RAGRating(StrEnum):
    GREEN = "green"
    AMBER = "amber"
    RED = "red"


class RegulatoryChangeType(StrEnum):
    NEW_LEGISLATION = "new_legislation"
    AMENDMENT = "amendment"
    NEW_REGULATORY_STANDARD = "new_regulatory_standard"
    CONSULTATION = "consultation"


class AlertStatus(StrEnum):
    PENDING = "pending"
    ACKNOWLEDGED = "acknowledged"
    DISMISSED = "dismissed"
    ACTIONED = "actioned"


class BillingEventType(StrEnum):
    CREDIT = "credit"
    USAGE = "usage"
    ADJUSTMENT = "adjustment"


class BillingFeature(StrEnum):
    CONVERSATION = "conversation"
    POLICY_REVIEW = "policy_review"
    TITLE_GENERATION = "title_generation"
    REGULATORY_MONITORING = "regulatory_monitoring"


class ActivityLogTag(StrEnum):
    USER = "user"
    SYSTEM = "system"
    SECURITY = "security"
    AI = "ai"


class FeedbackRating(StrEnum):
    UP = "up"
    DOWN = "down"


class FeedbackReviewStatus(StrEnum):
    PENDING = "pending"
    REVIEWED = "reviewed"
    ACTIONED = "actioned"

File: frontend/src/lib/types/enums.ts
/**
 * Enum types — Literal string unions matching backend StrEnums.
 *
 * Source of truth: API_CONTRACTS.md Section 6.1
 */

// Core
export type SubscriptionTier = "starter" | "professional" | "enterprise";
export type UserStatus = "pending" | "active" | "disabled" | "deleted";

// Conversations
export type ConversationState =
  | "pending"
  | "waiting_for_reply"
  | "generating_reply"
  | "outputting_reply"
  | "ready";
export type MessageRole = "user" | "assistant";
export type MessageState = "pending" | "success" | "error" | "cancelled";
export type ConfidenceLevel = "high" | "medium" | "low";

// Documents & Knowledge
export type KnowledgeBaseCategory =
  | "legislation"
  | "case_law"
  | "explanatory_notes"
  | "amendments"
  | "company_policy"
  | "sector_knowledge"
  | "parliamentary";
export type KnowledgeBaseSourceType =
  | "lex_api"
  | "uploaded"
  | "catalog"
  | "parliament_mcp";
export type DocumentProcessingState =
  | "uploaded"
  | "validating"
  | "extracting_text"
  | "chunking"
  | "contextualising"
  | "embedding"
  | "indexing"
  | "ready"
  | "failed";

// Verification & Citation
export type VerificationStatus =
  | "verified"
  | "unverified"
  | "removed"
  | "pre_1963_digitised";

// Agent
export type AgentInvocationMode = "conversation" | "policy_review";
export type ModelTier = "haiku" | "sonnet" | "opus";

// Guardrails
export type GuardrailStatus =
  | "creating"
  | "updating"
  | "versioning"
  | "ready"
  | "failed"
  | "deleting";

// Policy
export type PolicyReviewState =
  | "pending"
  | "processing"
  | "verifying"
  | "complete"
  | "error"
  | "cancelled";
export type ReviewCycle = "annual" | "monthly" | "quarterly";
export type RAGRating = "green" | "amber" | "red";
export type RegulatoryChangeType =
  | "new_legislation"
  | "amendment"
  | "new_regulatory_standard"
  | "consultation";
export type AlertStatus = "pending" | "acknowledged" | "dismissed" | "actioned";

// Billing
export type BillingEventType = "credit" | "usage" | "adjustment";
export type BillingFeature =
  | "conversation"
  | "policy_review"
  | "title_generation"
  | "regulatory_monitoring";

// Activity
export type ActivityLogTag = "user" | "system" | "security" | "ai";

// Feedback
export type FeedbackRating = "up" | "down";
export type FeedbackReviewStatus = "pending" | "reviewed" | "actioned";
